diff --git a/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/docker/ui/launch/ContainerLauncher.java b/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/docker/ui/launch/ContainerLauncher.java
index 4f47c3d..e38c500 100644
--- a/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/docker/ui/launch/ContainerLauncher.java
+++ b/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/docker/ui/launch/ContainerLauncher.java
@@ -101,13 +101,13 @@ public class ContainerLauncher {
 		final List<String> volumes = new ArrayList<>();
 		if (additionalDirs != null) {
 			for (String dir : additionalDirs) {
-				volumes.add(dir + ":" + dir); //$NON-NLS-1$
+				volumes.add(dir + ":" + dir + ":Z"); //$NON-NLS-1$ //$NON-NLS-2$
 			}
 		}
 		if (workingDir != null)
-			volumes.add(workingDir + ":" + workingDir); //$NON-NLS-1$
+			volumes.add(workingDir + ":" + workingDir + ":Z"); //$NON-NLS-1$ //$NON-NLS-2$
 		if (commandDir != null)
-			volumes.add(commandDir + ":" + commandDir); //$NON-NLS-1$
+			volumes.add(commandDir + ":" + commandDir + ":Z"); //$NON-NLS-1$ //$NON-NLS-2$
 
 		final List<String> cmdList = getCmdList(command);
 
diff --git a/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/wizards/ImageRun.java b/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/wizards/ImageRun.java
index f8009e1..43c12a2 100644
--- a/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/wizards/ImageRun.java
+++ b/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/wizards/ImageRun.java
@@ -156,9 +156,9 @@ public class ImageRun extends Wizard {
 			switch (dataVolume.getMountType()) {
 			case HOST_FILE_SYSTEM:
 				String bind = convertToUnixPath(dataVolume.getHostPathMount())
-						+ ':' + dataVolume.getContainerPath();
+						+ ':' + dataVolume.getContainerPath() + ':' + 'Z';
 				if (dataVolume.isReadOnly()) {
-					bind += ':' + "ro";
+					bind += ",ro"; //$NON-NLS-1$
 				}
 				binds.add(bind);
 				break;
-- 
1.7.1

