diff --git a/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/commands/BuildImageCommandHandler.java b/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/commands/BuildImageCommandHandler.java
index 1d87f4c..073b16a 100644
--- a/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/commands/BuildImageCommandHandler.java
+++ b/containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/commands/BuildImageCommandHandler.java
@@ -87,7 +87,10 @@ public class BuildImageCommandHandler extends AbstractHandler {
 								@Override
 								public FileVisitResult visitFile(
 										java.nio.file.Path file,
-										BasicFileAttributes attrs) {
+										BasicFileAttributes attrs) throws IOException {
+									if (!file.toFile().canRead()) {
+										throw new IOException();
+									}
 									return FileVisitResult.CONTINUE;
 								}
 								@Override
